backend/utils.py

```
@@ -0,0 +1,343 @@
+        image = Image.open(io.BytesIO(image_bytes)).convert('RGB')
+        
+        # Create input for Voyage AI (text + image pair)
+        inputs = [["DIY hardware item", image]]
+        
+        # Generate multimodal embedding
+        result = vo.multimodal_embed(inputs, model="voyage-multimodal-3")
+        
+        # Return the embedding vector
+        return result.embeddings[0]
+        
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)